pkg/mermaid/datamodeldiagram/datamodeldiagram.go

```diff
@@ -8,44 +8,101 @@ import (
 	"github.com/anz-bank/sysl/pkg/syslwrapper"
 )
 
-//cause linter..
-const ref = "ref"
+// Remove goconst linter warnings
+const (
+	ref      = "ref"
+	enum     = "enum"
+	mapS     = "map"
+	tuple    = "tuple"
+	relation = "relation"
+)
 
 //externalLink keeps track of the dependencies between two types
 type externalLink struct {
 	firstType, secondType string
 }
 
-//GenerateDataModelDiagram accepts a sysl module and returns a string (and an error if any)
+//GenerateFullDataDiagram accepts a sysl module and returns a string (and an error if any)
 //The resulting string is the mermaid code for the data model for that module
-func GenerateDataModelDiagram(m *sysl.Module) (string, error) {
+func GenerateFullDataDiagram(m *sysl.Module) (string, error) {
+	mapper := syslwrapper.MakeAppMapper(m)
+	mapper.IndexTypes()
+	mapper.ConvertTypes()
+	return generateFullDataDiagramHelper(mapper.SimpleTypes, &[]externalLink{})
+}
+
+//GenerateDataDiagramWithAppAndType generates the data model for a specific type
+func GenerateDataDiagramWithAppAndType(m *sysl.Module, appName string, typeName string) (string, error) {
 	mapper := syslwrapper.MakeAppMapper(m)
 	mapper.IndexTypes()
 	mapper.ConvertTypes()
-	return generateDataModelDiagramHelper(mapper.SimpleTypes, &[]externalLink{})
+	return generateDataDiagramWithAppAndTypeHelper(mapper.SimpleTypes, appName, typeName,
+		&[]externalLink{}, &[]string{}, &[]externalLink{})
+}
+
+//generateDataDiagramWithAppAndTypeHelper is a helper which has additional arguments
+func generateDataDiagramWithAppAndTypeHelper(simpleTypes map[string]*syslwrapper.Type, appName string,
+	typeName string, externalLinks *[]externalLink, appTypes *[]string, eLinks *[]externalLink) (string, error) {
+	var result string
+	appType1 := fmt.Sprintf("%s:%s", appName, typeName)
+	result = mermaid.GeneratedHeader + "classDiagram\n"
+	for appType, value := range simpleTypes {
```

Comment:
Looks like this for loop is duplicated at line 89? (I might've missed a difference). Could this be pulled out into a function, that gets reused between generateDataDiagramWithAppAndTypeHelper and generateFullDataDiagramHelper?

Reply:
Just a tiny difference, I reused the same to get all the possible links between types. That one doesn't actually build the string.